app/components/edit-reminder.js

```diff
@@ -24,6 +24,11 @@ export default Ember.Component.extend({
         record.save();
       })
     },
+    reset(){
+      this.element.children[0].textContent = this.model.data.title;
```

Comment:
Check out [the docs](http://emberjs.com/api/data/classes/DS.Model.html#method_rollbackAttributes) to do this in an easier way